test/test_jit.py

```diff
@@ -7592,6 +7593,26 @@ def foo():
                     raise Exception("Hi")
                 return a
 
+    def test_assertions(self):
+        cu = torch.jit.CompilationUnit('''
+            def foo(cond):
+                assert bool(cond), "hi"
+                return 0
+        ''')
+
+        cu.foo(torch.tensor(1))
+        with self.assertRaisesRegex(torch.jit._Exception, "Exception"):
```

Comment:
unrelated but this should probably be `JITException` or something similar since `_Exception` is going to be exposed to users

Reply:
Yea agreed _Exception isn't great. Maybe torch.jit.Error, torch.JITException ?

Reply:
`...Error` is the [Python convention](https://www.python.org/dev/peps/pep-0008/#exception-names)